### PR TITLE
FIX: [droid] Only sw for SD MPEG

### PR DESCRIPTION
**Pull request by koying**
Don't bother with SD MPEG peculiarities.
Fixes VOB playback

### CODE REVIEW

File: xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp [hunk 1]
```diff
@@ -216,6 +216,8 @@ CDVDVideoCodec* CDVDFactoryCodec::CreateVideoCodec(CDVDStreamInfo &hint, const C
       case AV_CODEC_ID_MPEG4:
       case AV_CODEC_ID_MSMPEG4V2:
       case AV_CODEC_ID_MSMPEG4V3:
+      case AV_CODEC_ID_MPEG1VIDEO:
+      case AV_CODEC_ID_MPEG2VIDEO:
```
Comment (FernetMenta) [inline xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp]: why don't you do those checks in the codec?

Comment (koying) [inline xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp]: Seems clearer to me to centralize those "support reducing exceptions" here.

Comment (FernetMenta) [inline xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp]: But we should follow the principle of information hiding that we already apply to the ffmpeg hw accelerators. Example:
https://github.com/xbmc/xbmc/blob/master/xbmc/cores/dvdplayer/DVDCodecs/Video/VAAPI.cpp#L532

Platform independent code should no be touched if something in the dependent code changes. Finally ifdefs should go away as well. IMO we should consider this next time this code is touched.

Comment (FernetMenta) [inline xbmc/cores/dvdplayer/DVDCodecs/DVDFactoryCodec.cpp]: this is btw some part of the code where I agree to @jimfcarroll 's di. platform dependent code could register a decoder here and factory is free of ifdefs.

**Pull request merged**